client/src/client.rs

```diff
@@ -986,13 +986,22 @@ impl RpcApi for Client {
     ) -> Result<T> {
         let req = self.client.build_request(&cmd, &args);
         if log_enabled!(Debug) {
-            debug!("JSON-RPC request: {}", serde_json::to_string(&req).unwrap());
+            debug!(target: "bitcoincore_rpc", "JSON-RPC request: {} {}", cmd, serde_json::Value::from(args));
         }
 
         let resp = self.client.send_request(&req).map_err(Error::from);
-        if log_enabled!(Debug) && resp.is_ok() {
-            let resp = resp.as_ref().unwrap();
-            debug!("JSON-RPC response: {}", serde_json::to_string(resp).unwrap());
+        if log_enabled!(Warn) {
+            match resp {
+                Err(ref e) => warn!(target: "bitcoincore_rpc", "JSON-RPC error for {}: {:?}", cmd, e),
+                Ok(ref resp) => {
+                    if let Some(ref e) = resp.error {
+                        warn!(target: "bitcoincore_rpc", "JSON-RPC error for {}: {:?}", cmd, e);
+                    } else if log_enabled!(Trace) {
+                        let result = resp.result.as_ref().unwrap_or(&serde_json::Value::Null);
+                        trace!(target: "bitcoincore_rpc", "JSON-RPC response for {}: {}", cmd, result);
```

Comment:
This trace is under the `if log_enabled!(Warn)` so it won't be logged unless warn is enabled.

Reply:
Hmm, yes, you're correct. I typically use the "level filters" which also enable everything below the specified level (so `Warn` would also enable`Info`, `Debug` and `Trace`), but it isn't necessary has to be used like that. Will change this.

Reply:
Oh, yeah you're right it might work like that. That might be my bad. I think that's how it works in the log crate. Nvm